odl/solvers/util/callback.py

```diff
@@ -438,19 +440,137 @@ def reset(self):
         """Set `iter` to 0 and create a new figure."""
         self.iter = 0
         self.fig = None
+        self.space_of_last_x = None
 
     def __repr__(self):
         """Return ``repr(self)``."""
-
         return '{}(display_step={}, saveto={}, fig={}, *{!r}, **{!r})'.format(
             self.__class__.__name__,
-            self.saveto,
             self.display_step,
+            self.saveto,
             self.fig,
             self.args,
             self.kwargs)
 
 
+class CallbackSaveToDisk(SolverCallback):
+
+    """Save the iterates to disk."""
+
+    def __init__(self, saveto, save_step=1, impl='pickle', **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        saveto : string
+            Path to a directory where the figures are to be saved, followed by
+            a file name and potentially a file format. Standard python string
+            formatting is possible to use in order to index the iterates with
+            iteration number.
```

Comment:
This could be clearer here already, before it gets clarified in the examples. Here's a suggestion:
```
saveto : string
    Format string for the name of the file(s) where
    iterates are saved. The file name is generated as

        filename = saveto.format(cur_iter_num)

    where ``cur_iter_num`` is the current iteration number.
```
At least to me, that is much clearer than all the text.